libqtile/widget/mpdwidget.py

```diff
@@ -30,285 +30,270 @@
+            else:
+                return self.can_connect(is_retry=True)
 
-        self.connected = True
         return True
 
-    def _configure(self, qtile, bar):
-        super()._configure(qtile, bar)
-        self.layout = self.drawer.textlayout(
-            self.text,
-            self.foreground,
-            self.font,
-            self.fontsize,
-            self.fontshadow,
-            markup=True
-        )
-
-    def to_minutes_seconds(self, stime):
-        """Takes an integer time in seconds, transforms it into
-        (HH:)?MM:SS. HH portion is only visible if total time is greater
-        than an hour.
-        """
-        if type(stime) != int:
-            stime = int(stime)
-        mm = stime // 60
-        ss = stime % 60
-        if mm >= 60:
-            hh = mm // 60
-            mm = mm % 60
-            rv = "{}:{:02}:{:02}".format(hh, mm, ss)
-        else:
-            rv = "{}:{:02}".format(mm, ss)
-        return rv
-
-    def get_artist(self):
-        return self.song['artist']
-
-    def get_album(self):
-        return self.song['album']
-
-    def get_elapsed(self):
-        elapsed = self.status['time'].split(':')[0]
-        return self.to_minutes_seconds(elapsed)
-
-    def get_file(self):
-        return self.song['file']
-
-    def get_length(self):
-        return self.to_minutes_seconds(self.song['time'])
-
-    def get_number(self):
-        return str(int(self.status['song']) + 1)
-
-    def get_playlistlength(self):
-        return self.status['playlistlength']
-
-    def get_status(self):
-        n = self.status['state']
-        if n == "play":
-            return "->"
-        elif n == "pause":
-            return "||"
-        elif n == "stop":
-            return "[]"
-
-    def get_longstatus(self):
-        n = self.status['state']
-        if n == "play":
-            return "Playing"
-        elif n == "pause":
-            return "Paused"
-        elif n == "stop":
-            return "Stopped"
-
-    def get_title(self):
-        return self.song['title']
-
-    def get_track(self):
-        # This occasionally has leading zeros we don't want.
-        return str(int(self.song['track'].split('/')[0]))
-
-    def get_volume(self):
-        return self.status['volume']
-
-    def get_single(self):
-        if self.status['single'] == '1':
-            return '1'
+    def poll(self):
+        if self.can_connect():
+            return self.update_status()
         else:
-            return '_'
+            return self.no_connection
 
-    def get_repeat(self):
-        if self.status['repeat'] == '1':
-            return 'R'
+    def button_press(self, x, y, button, is_retry=False):
+        if self.can_connect(is_retry=is_retry):
+            try:
+                self.handle_button_press(x, y, button)
+                self.update(self.update_status())
+            except (CommandError, BrokenPipeError, ProtocolError) as e:
+                logger.warning(f'mpd error: {e}')
```

Comment:
Seems like a syntax error here that travis is choking on it?

Reply:
```
libqtile/widget/mpdwidget.py:206:` error: Format strings are only supported in Python 3.6 and greater

Found 1 error in 1 file (checked 124 source files)

ERROR: InvocationError for command /home/travis/build/qtile/qtile/.tox/mypy/bin/mypy -p libqtile (exited with code 2)

```

That's why I didn't use them in the mpd2 widget.

Reply:
Thanks for pointing this out!